sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/InMemoryCatalog.scala

```diff
@@ -145,7 +145,7 @@ class InMemoryCatalog(
         fs.delete(location, true)
       } catch {
         case e: IOException =>
-          throw new SparkException(s"Unable to drop database ${dbDefinition.name} as failed " +
+          throw new IOException(s"Unable to drop database ${dbDefinition.name} as failed " +
```

Comment:
Not related to this PR, but just as a note I think it would be more grammatically correct to say "since deletion of ${..} failed" rather than "as we failed to delete ${..}".